argilla/docs/how_to_guides/query.md

```diff
@@ -63,6 +63,35 @@ To search for records with terms, you can use the `Dataset.records` attribute wi
     queried_records = dataset.records(query=query).to_list(flatten=True)
     ```
 
+### Advanced queries
+
+If you need more complex searches, you can use [Elasticsearch's simple query string syntax](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-simple-query-string-query.html#simple-query-string-syntax). Here is a summary of the different available operators:
+
+=== "Descriptions"
+    | operator     | description                                               |
+    | ------------ | --------------------------------------------------------- |
+    |`+` or `space`| **AND**: matches records that contain both terms          |
+    |`|`           | **OR**: matches records that contain either term          |
+    |`-`           | **Negation**: exclude records that contain a specific term|
+    |`*`           | **Prefix**:matches records that contain a specific prefix |
+    |`"`           | **Phrase**: matches records that contain a specific phrase|
+    |`(` and `)`   | **Precedence**: group terms                               |
+    |`~N`          | **Edit distance**: matches records that contain a term or phrase with an edit distance|
+
+=== "Examples"
+    | operator     |  example                      | explanation                                     |
+    | ------------ | ----------------------------- | ----------------------------------------------- |
+    |`+` or `space`| `argilla + distilabel` or `argilla distilabel`| returns records that include the terms "argilla" and "distilabel"|
+    |`|`           | `argilla | distilabel`        | returns records that include the term "argilla" or "distilabel"|
+    |`-`           | `argilla -distilabel`         | returns records that contain the term "argilla" and don't have the term "distilabel"|
+    |`*`           | `arg*`                        | returns records with any words starting with "arg-"|
+    |`"`           | `"argilla and distilabel"`    | returns records that contain the phrase "argilla and distilabel"|
+    |`(` and `)`   | `(argilla | distilabel) rules`| returns records that contain either "argilla" or "distilabel" and "rules"|
+    |`~N`          | `argilla~1`                   | returns records that contain the term "argilla" with an edit distance of 1, e.g. "argila"|
+
```

Comment:
What do you think about using a single table? I feel having to switch between tabs to see the examples is uncomfortable, and I think using one keeps it structured, readable, and is not too big either.

<img width="1490" alt="Screenshot 2024-08-29 at 12 32 51" src="https://github.com/user-attachments/assets/706feed1-2215-412b-902a-f2b7e4ca3895">

Reply:
Thanks, @sdiazlor! I thought it might take too much space, but it looks fine as you show it there. I'll make the change!

Reply:
Here's how it would look. The problem is that some examples aren't very clear because of the available space. For example: 
- in the second example in the first row it's not easy to see that there is a space between `argilla distilabel`.
- in the third row, the `-` should be next to the term it negates, as in `-distilabel`, but because of the line break this is not very obvious... 
Any thoughts?
![Captura de pantalla 2024-08-29 a las 13 10 28](https://github.com/user-attachments/assets/0aea1e27-8f87-426b-ad1f-5d55fe786081)